dashboard/modules/job/job_manager.py

```diff
@@ -206,13 +206,20 @@ async def run(
                 "runtime_env": self._runtime_env,
                 "metadata": self._metadata,
             })
-            # Set RAY_ADDRESS to local Ray address, if it is not set.
+            # Always set RAY_ADDRESS as redis address for job submission.
+            # In case of local development, prevent user from re-using
+            # http://{address}:{dashboard_port} to interact with jobs SDK.
             os.environ[
                 ray_constants.RAY_ADDRESS_ENVIRONMENT_VARIABLE] = \
-                ray._private.services.get_ray_address_from_environment()
+                ray._private.services.find_redis_address().pop()
```

Comment:
ray._private.services.find_bootstrap_address().pop()

And you can add a TODO for me to check why "auto" does not work.